### PR TITLE
Get an additional 10% improvement for NavTo

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Followup to https://github.com/dotnet/roslyn/pull/20602

Features a much simpler string breaker

### CODE REVIEW

File: src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.InProcess.cs [hunk 1]
```diff
@@ -199,9 +204,15 @@ internal abstract partial class AbstractNavigateToSearchService
             var kind = GetItemKind(declaredSymbolInfo);
             var navigableItem = NavigableItemFactory.GetItemFromDeclaredSymbolInfo(declaredSymbolInfo, document);
 
+            var matchedSpans = ArrayBuilder<TextSpan>.GetInstance();
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.InProcess.cs]: handwritten to avoid linq allocation.

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_unit64_prtest please

**Comment by CyrusNajmabadi** [comment]
retest windows_release_vs-integration_prtest please

File: src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.InProcess.cs [hunk 2]
```diff
@@ -60,9 +61,13 @@ internal abstract partial class AbstractNavigateToSearchService
                     // use the last computed results we have for that project.  If so, it can
                     // be much faster to reuse and filter that result than to compute it from
                     // scratch.
+#if true
                     var task = searchDocument != null
                         ? ComputeSearchResultsAsync(project, searchDocument, nameMatcher, containerMatcherOpt, nameMatches, containerMatches, cancellationToken)
                         : TryFilterPreviousSearchResultsAsync(project, searchDocument, pattern, nameMatcher, containerMatcherOpt, nameMatches, containerMatches, cancellationToken);
+#else
```
Comment (rchande) [inline src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.InProcess.cs]: huh?

File: src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs [hunk 3]
```diff
@@ -3,6 +3,7 @@
 using System;
 using System.Collections.Generic;
 using System.Collections.Immutable;
+using System.Diagnostics;
```
Comment (rchande) [inline src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs]: Unused?

File: src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.InProcess.cs (hunk 2 above)
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.InProcess.cs]: I've been perf testing.  It's an easy way to disable some caching behavior so i can just profile the speed of matching.

**Comment by CyrusNajmabadi** [comment]
Closing in favor of https://github.com/dotnet/roslyn/pull/20847

**Pull request closed without merging**